packages/autocomplete-core/src/__tests__/getInputProps.test.ts

```diff
@@ -1165,9 +1165,151 @@ describe('getInputProps', () => {
     });
   });
 
-  describe('onFocus', () => {});
+  describe('onFocus', () => {
+    test('triggers a query if `openOnFocus` is true', () => {
+      const getSources = jest.fn(() => [createSource()]);
+      const { inputElement } = createPlayground(createAutocomplete, {
+        openOnFocus: true,
+        getSources,
+      });
+
+      inputElement.focus();
+
+      expect(getSources).toHaveBeenCalledTimes(1);
+    });
+
+    test('triggers a query if the current query is not empty', () => {
+      const getSources = jest.fn(() => [createSource()]);
+      const { inputElement } = createPlayground(createAutocomplete, {
+        getSources,
+        initialState: {
+          query: 'i',
+        },
+      });
+
+      inputElement.focus();
+
+      expect(getSources).toHaveBeenCalledTimes(1);
+    });
+
+    test('sets activeItemId and isOpen', () => {
```

Comment:
We should test these separately, they test the behavior of the state reducer:

- sets default activeItemId
  - → assert that `activeItemId === null`
- sets default activeItemId with defaultActiveItemId with 0
  - → assert that `activeItemId === 0`
- sets isOpen without query
  - → assert that `isOpen === false`
- sets isOpen with query
  - → assert that `isOpen === true`
- sets isOpen with openOnFocus
  - → assert that `isOpen === true`
- sets isOpen with openOnFocus and query
  - → assert that `isOpen === true`